### PR TITLE
remove config-trusted-resources

### LINKED ISSUE #5852: Deprecate config-trusted-resources when VerificationPolicy is supported
[config-trusted-resources](https://github.com/tektoncd/pipeline/blob/main/config/config-trusted-resources.yaml) is introduced for trusted resources to store public key reference, the feature will be fully covered by [VerificationPolicy](https://github.com/tektoncd/pipeline/pull/5714) and should be deprecated in future releases. 

/kind api-change

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit removes config-trusted-resources. The deprecation is announced in release v0.45. The reason of removing is that config-trusted-resources is used to store public keys for verificaiton but Verification Policy has already covered all the functionalities and has more advanced features. Since there are not any other fields in trusted-resources-config we decided to remove it. Trusted resources is alpha feature so the configmap can be deprecated with one release notice

Closes #5852

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
BREAKING CHANGE: [alpha] config-trusted-resources is removed, please refer to https://github.com/tektoncd/pipeline/blob/main/docs/trusted-resources.md for migrating public keys in VerificationPolicy
```

### CODE REVIEW

File: config/controller.yaml [hunk 1]
```diff
@@ -88,10 +88,6 @@ spec:
           mountPath: /etc/config-logging
         - name: config-registry-cert
           mountPath: /etc/config-registry-cert
-        # Mount secret for trusted resources
-        - name: verification-secrets
-          mountPath: /etc/verification-secrets
-          readOnly: true
```
Comment (Yongxuanzhang) [inline config/controller.yaml:94]: This was introduced to mount secrets(contains public keys) and configmap could refer to the secret.

File: test/trusted_resources_test.go [hunk 2]
```diff
@@ -55,177 +55,12 @@ func init() {
 	os.Setenv("PRIVATE_PASSWORD", password)
 }
 
-func TestTrustedResourcesVerify_ConfigMap_Success(t *testing.T) {
```
Comment (Yongxuanzhang) [inline test/trusted_resources_test.go:58]: This is the e2e test of configmap approach.

**Review by lbernick — APPROVED** [review]
thanks @Yongxuanzhang, please update the release note to include more detail (including a link to where people can find migration instructions) and update the commit message to note that this is an alpha feature.

File: pkg/trustedresources/errors.go [hunk 3]
```diff
@@ -20,8 +20,8 @@ import "errors"
 var (
 	// ErrorResourceVerificationFailed is returned when trusted resources fails verification.
 	ErrorResourceVerificationFailed = errors.New("resource verification failed")
-	// ErrorEmptyVerificationConfig is returned when VerificationPolicy or config-trusted-resources configmap are founded
-	ErrorEmptyVerificationConfig = errors.New("no policies or config-trusted-resources configmap founded for verification")
+	// ErrorEmptyVerificationConfig is returned when VerificationPolicy are founded
```
Comment (wlynch) [inline pkg/trustedresources/errors.go:23]:
```suggestion
	// ErrorEmptyVerificationConfig is returned when VerificationPolicy are found
```

**Review by wlynch — APPROVED** [review]
/lgtm

**Review by wlynch — COMMENTED** [review]
/lgtm

**Pull request merged**